src/main.rs

```diff
@@ -78,36 +79,54 @@ async fn main() -> Result<()> {
+
+                // verify policy prior to pulling if keys listed, and keep the
+                // verified manifest digest of last iteration, even if all are
+                // the same:
+                let mut verified_manifest_digest: String = String::new();
+                for key_file in key_files.iter() {
+                    // let key = verify::read_key_file(&key_file[0])?;
+                    verified_manifest_digest = verify::verify(
+                        uri,
+                        docker_config.clone(),
+                        sources.clone(),
+                        annotations.clone(),
+                        &key_file[0],
+                    )
+                    .await?;
+                }
+                pull::pull(uri, docker_config.clone(), sources.clone(), destination).await?;
+                if !verified_manifest_digest.is_empty() {
+                    verify::verify_local_checksum(
+                        uri,
+                        docker_config,
+                        sources,
+                        &verified_manifest_digest,
+                    )
+                    .await?;
+                }
             };
             Ok(())
         }
         Some("verify") => {
             if let Some(matches) = matches.subcommand_matches("verify") {
                 let uri = matches.value_of("uri").unwrap();
-                let key_file = matches.value_of("key").unwrap();
                 let (sources, docker_config) = remote_server_options(matches)?;
+                let (key_files, annotations) = verification_options(matches)?;
 
-                let annotations: Option<HashMap<String, String>>;
-                annotations = match matches.values_of("annotations") {
-                    None => None,
-                    Some(items) => {
-                        let mut values: HashMap<String, String> = HashMap::new();
-                        for item in items {
-                            let tmp: Vec<_> = item.splitn(2, '=').collect();
-                            if tmp.len() == 2 {
-                                values.insert(String::from(tmp[0]), String::from(tmp[1]));
-                            }
-                        }
-                        if values.is_empty() {
-                            None
-                        } else {
-                            Some(values)
-                        }
-                    }
-                };
-
-                verify::verify(uri, docker_config, sources, annotations, key_file).await?;
+                for key_file in key_files.iter() {
+                    println!("{:}", &key_file[0]);
+                    // let key = verify::read_key_file(&key_file[0])?;
+                    verify::verify(
+                        uri,
+                        docker_config.clone(),
+                        sources.clone(),
+                        annotations.clone(),
```

Comment:
Same here, I would pass all of them by reference

Reply:
Fixed, thanks!